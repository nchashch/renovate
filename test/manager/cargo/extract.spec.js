const fs = require('fs');
const { extractPackageFile } = require('../../../lib/manager/cargo/extract');

const cargo1toml = fs.readFileSync('test/_fixtures/cargo/Cargo.1.toml', 'utf8');
const cargo2toml = fs.readFileSync('test/_fixtures/cargo/Cargo.2.toml', 'utf8');
const cargo3toml = fs.readFileSync('test/_fixtures/cargo/Cargo.3.toml', 'utf8');

describe('lib/manager/cargo/extract', () => {
  describe('extractPackageFile()', () => {
    let config;
    beforeEach(() => {
      config = {};
    });
    it('returns null for empty', () => {
      expect(extractPackageFile('nothing here', config)).toBe(null);
    });
    it('extracts multiple dependencies', () => {
      const res = extractPackageFile(cargo1toml, config);
      // expect(res.deps).toMatchSnapshot();
      expect(res.deps).toHaveLength(6);
    });
    it('extracts multiple dependencies', () => {
      const res = extractPackageFile(cargo2toml, config);
      // expect(res.deps).toMatchSnapshot();
      expect(res.deps).toHaveLength(18 + 6 + 1);
    });
    it('extracts handles inline tables', () => {
      const res = extractPackageFile(cargo3toml, config);
      // expect(res.deps).toMatchSnapshot();
      expect(res.deps).toHaveLength(6);
    });
  });
});
