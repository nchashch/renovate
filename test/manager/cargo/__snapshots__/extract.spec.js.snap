// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/manager/cargo/extract extractPackageFile() extracts multiple dependencies 1`] = `
Array [
  Object {
    "currentValue": "0.2.43",
    "depName": "libc ",
    "depType": "normal",
    "lineNumber": 16,
  },
  Object {
    "currentValue": "1.0.4",
    "depName": "bitflags ",
    "depType": "normal",
    "lineNumber": 17,
  },
  Object {
    "currentValue": "0.1",
    "depName": "pcap-sys ",
    "depType": "inline-table",
    "lineNumber": 18,
  },
  Object {
    "currentValue": "0.21.0",
    "depName": "pnet ",
    "depType": "inline-table",
    "lineNumber": 19,
  },
  Object {
    "currentValue": "0.3.6",
    "depName": "winapi",
    "depType": "standard-table",
    "lineNumber": 21,
    "versionLineNumber": 22,
  },
]
`;

exports[`lib/manager/cargo/extract extractPackageFile() extracts multiple dependencies 2`] = `
Array [
  Object {
    "currentValue": "0.2.0",
    "depName": "amethyst_animation ",
    "depType": "inline-table",
    "lineNumber": 53,
  },
  Object {
    "currentValue": "0.3.0",
    "depName": "amethyst_assets ",
    "depType": "inline-table",
    "lineNumber": 54,
  },
  Object {
    "currentValue": "0.2.0",
    "depName": "amethyst_audio ",
    "depType": "inline-table",
    "lineNumber": 55,
  },
  Object {
    "currentValue": "0.6.0",
    "depName": "amethyst_config ",
    "depType": "inline-table",
    "lineNumber": 56,
  },
  Object {
    "currentValue": "0.2.0",
    "depName": "amethyst_core ",
    "depType": "inline-table",
    "lineNumber": 57,
  },
  Object {
    "currentValue": "0.1.0",
    "depName": "amethyst_controls ",
    "depType": "inline-table",
    "lineNumber": 58,
  },
  Object {
    "currentValue": "0.1.0",
    "depName": "amethyst_locale ",
    "depType": "inline-table",
    "lineNumber": 59,
  },
  Object {
    "currentValue": "0.7",
    "depName": "amethyst_renderer ",
    "depType": "inline-table",
    "lineNumber": 60,
  },
  Object {
    "currentValue": "0.3",
    "depName": "amethyst_input ",
    "depType": "inline-table",
    "lineNumber": 61,
  },
  Object {
    "currentValue": "0.2",
    "depName": "amethyst_ui ",
    "depType": "inline-table",
    "lineNumber": 62,
  },
  Object {
    "currentValue": "0.2",
    "depName": "amethyst_utils ",
    "depType": "inline-table",
    "lineNumber": 63,
  },
  Object {
    "currentValue": "1.0",
    "depName": "derivative ",
    "depType": "normal",
    "lineNumber": 64,
  },
  Object {
    "currentValue": "0.5",
    "depName": "fern ",
    "depType": "inline-table",
    "lineNumber": 65,
  },
  Object {
    "currentValue": "0.4",
    "depName": "log ",
    "depType": "normal",
    "lineNumber": 66,
  },
  Object {
    "currentValue": "1.0.1",
    "depName": "rayon ",
    "depType": "normal",
    "lineNumber": 67,
  },
  Object {
    "currentValue": "0.1",
    "depName": "rustc_version_runtime ",
    "depType": "normal",
    "lineNumber": 68,
  },
  Object {
    "currentValue": "0.15",
    "depName": "winit ",
    "depType": "normal",
    "lineNumber": 69,
  },
  Object {
    "currentValue": "0.1",
    "depName": "thread_profiler ",
    "depType": "inline-table",
    "lineNumber": 71,
  },
  Object {
    "currentValue": "0.2",
    "depName": "amethyst_gltf ",
    "depType": "inline-table",
    "lineNumber": 74,
  },
  Object {
    "currentValue": "0.5.10",
    "depName": "env_logger ",
    "depType": "normal",
    "lineNumber": 75,
  },
  Object {
    "currentValue": "0.6",
    "depName": "genmesh ",
    "depType": "normal",
    "lineNumber": 76,
  },
  Object {
    "currentValue": "0.2",
    "depName": "ron ",
    "depType": "normal",
    "lineNumber": 77,
  },
  Object {
    "currentValue": "1.0",
    "depName": "serde ",
    "depType": "normal",
    "lineNumber": 78,
  },
  Object {
    "currentValue": "1.0",
    "depName": "serde_derive ",
    "depType": "normal",
    "lineNumber": 79,
  },
  Object {
    "currentValue": "0.1",
    "depName": "vergen ",
    "depType": "normal",
    "lineNumber": 82,
  },
]
`;

exports[`lib/manager/cargo/extract extractPackageFile() handles inline tables 1`] = `
Array [
  Object {
    "currentValue": "0.1",
    "depName": "pcap-sys ",
    "depType": "inline-table",
    "lineNumber": 5,
  },
  Object {
    "currentValue": "0.21.0",
    "depName": "pnet ",
    "depType": "inline-table",
    "lineNumber": 6,
  },
  Object {
    "currentValue": "1.2",
    "depName": "dep1 ",
    "depType": "inline-table",
    "lineNumber": 7,
  },
  Object {
    "currentValue": "3.4",
    "depName": "dep2 ",
    "depType": "inline-table",
    "lineNumber": 8,
  },
  Object {
    "currentValue": "~12.3.1",
    "depName": "dep3 ",
    "depType": "inline-table",
    "lineNumber": 9,
  },
  Object {
    "currentValue": "INVALID 3.3.1 VERSION",
    "depName": "dep4 ",
    "depType": "inline-table",
    "lineNumber": 10,
    "skipReason": "unknown-version",
  },
]
`;

exports[`lib/manager/cargo/extract extractPackageFile() handles standard tables 1`] = `
Array [
  Object {
    "currentValue": "1.2",
    "depName": "dep1",
    "depType": "standard-table",
    "lineNumber": 4,
    "versionLineNumber": 8,
  },
  Object {
    "currentValue": "3.4",
    "depName": "dep2",
    "depType": "standard-table",
    "lineNumber": 13,
    "versionLineNumber": 17,
  },
  Object {
    "currentValue": "~12.3.1",
    "depName": "dep3",
    "depType": "standard-table",
    "lineNumber": 18,
    "versionLineNumber": 19,
  },
  Object {
    "currentValue": "INVALID 1.3.1 VERSION",
    "depName": "dep4",
    "depType": "standard-table",
    "lineNumber": 23,
    "skipReason": "unknown-version",
    "versionLineNumber": 24,
  },
]
`;
